requests/adapters.py

```diff
@@ -299,8 +300,11 @@ def get_connection(self, url, proxies=None):
         proxy = select_proxy(url, proxies)
 
         if proxy:
-            proxy = prepend_scheme_if_needed(proxy, 'http')
-            proxy_manager = self.proxy_manager_for(proxy)
+            _proxy = prepend_scheme_if_needed(proxy, 'http')
+            proxy_url = parse_url(_proxy)
+            if not proxy_url.host:
+                raise ProxyError("URL for proxy is invalid: {}".format(proxy))
```

Comment:
This is really not a helpful message. Instead it should probably say that the proxy URL was missing the host and cannot be used. The URL should be checked. We likely also do not want to include the proxy URL in the exception message because that my have sensitive credential information and we don't want to accidentally leak that on behalf of the user

Reply:
Updated the error message.